### DISCUSSION TITLE
Outputted list length

### DISCUSSION THREAD

**Discussion by Woisek** [oobabooga/text-generation-webui#2778] [Q&A]
I'm currently playing with a vicuna model to use it for all kinds of questions. Now I've notice a thing and I want to know what I have/should do.
When I ask the bot for a list, like "write me ... one per line", then I get the lines perfectly, BUT at some time it stops the output somewhere in the middle.
My question is: is this avoidable? Is this controllable? Is there a setting that can be used?

And also a itching question is: What chat settings should be entered, if one like to make a "specialized bot" for some specific topic?
I know from using Pygmalion models, that there is a specific kind of template is needed, a way on how the fields are populated like Character's name, Context and Greeting.
I guess this applies also to "instruction(-chat)s", but I haven't found any docs to read about it.
It would be nice if someone could point me in the right direction, thanks! 🙂

**Comment by BetaDoggo** [comment]
You probably need to increase the max number of new tokens to generate. It should stop once it has completed its response anyway so you can set it to 2000. If it still stops after that it could be an issue with the model itself outputting an EOS token prematurely. Enabling "Ban EOS Token" in the settings should fix it, though you will have to set a stopping string so that it doesn't talk endlessly.

About specialized bots, Vicuna was trained with the prompt format:
```
A chat between a curious user and an artificial intelligence assistant. The assistant gives helpful, detailed, and polite answers to the user's questions.

USER: prompt
ASSISTANT:
```
Rewriting the persona given in the initial prompt might guide the model to make more detailed answers about a specific topic.

**Reply by Woisek** [reply]
Thanks for your reply. I will try the tip with the tokens, sounds logic to me.
As for your second answer, I'm sorry I have to re-questioning this. Where exactly would your example go?
Somewhere to User string, Bot string, Context?
And with _Rewriting the persona given in the initial prompt_ you mean the context field above?

Sorry for the maybe annoying questions, I'm not that dumb (I hope 😋), but I somehow don't find a page where all this entries in oobabooba are described, let alone finding examples or tutorials. for this kind of things.
I really want to get into this, it's a very interesting field. Thanks again for giving help. 👍
